blockchain/state_manager/src/chain_rand.rs

```diff
@@ -25,16 +26,18 @@ impl<DB> ChainRand<DB> {
 
 impl<DB> Rand for ChainRand<DB>
 where
-    DB: BlockStore,
+    DB: BlockStore + Send + Sync + 'static,
 {
     fn get_chain_randomness(
         &self,
         pers: DomainSeparationTag,
         round: ChainEpoch,
         entropy: &[u8],
     ) -> Result<[u8; 32], Box<dyn Error>> {
-        self.cs
-            .get_chain_randomness(&self.blks, pers, round, entropy)
+        task::block_on(
```

Comment:
We dont want to just make this an async_trait? Is it because this is really only used in the VM?

Reply:
yeah, the vm execution is sync, and I don't want to introduce async primitives when there will never be any benefit to it